common/src/main/java/org/apache/rocketmq/common/BrokerConfig.java

```diff
@@ -47,6 +47,7 @@
     private boolean autoCreateSubscriptionGroup = true;
     private String messageStorePlugIn = "";
 
+    //thread counts for sending message, if large number threads are needed, we suggest setting useReentrantLockWhenPutMessage=true(in MessageStoreConfig)
```

Comment:
@Jaskey , would you carry out an experiment, figuring out when spin-lock is better and when the alternative mutex-lock wins out? 

You may compose an article based on that experiment, which should be added to [best practice](http://rocketmq.incubator.apache.org/docs/best-practice-broker/) section of RocketMQ official website.  It's currently vacant!